src/Models/Scopes/Product/WithProductStockScope.php

```diff
@@ -24,7 +24,7 @@ public function apply(Builder $builder, Model $model)
         // - use_config_qty_increments
 
         $builder
-            ->selectRaw('ANY_VALUE(IF(cataloginventory_stock_item.use_config_backorders, ' . $configBackorder . ', cataloginventory_stock_item.backorders)) as backorder_type')
+            ->selectRaw('ANY_VALUE(IF(cataloginventory_stock_item.use_config_backorders, ' . ($configBackorder ?: '0') . ', cataloginventory_stock_item.backorders)) as backorder_type')
```

Comment:
This one is kind of stupid. The correct magento default is an empty string, however we can't put an empty string into the query. Hence this fallback.